routes/api/contacts.routes.js

```diff
@@ -13,94 +13,201 @@ const schema = Joi.object({
+      res.status(404).json({
+        status: "not-found",
+        code: 404,
+        message: "Not found",
+      });
+    }
+
+    res.json({
+      status: "success",
+      code: 200,
+      data: {
+        contact: contactToFind,
+      },
+    });
+  } catch (error) {
+    res.status(400).json({
+      status: "error",
+      code: 400,
       message: error.message,
     });
   }
+});
 
-  if (!name || !email || !phone) {
-    return res.status(400).json({
-      message: "Missing field",
+router.post("/", async (req, res, next) => {
+  try {
+    const { error, value } = schema.validate(req.body);
+    const { name, email, phone } = value;
+
+    if (error) {
+      res.status(400).json({
+        status: "error",
+        code: 400,
+        message: error.message,
+      });
+    }
+
+    if (!name || !email || !phone) {
+      res.status(400).json({
+        status: "error",
+        code: 400,
+        message: "Missing field",
+      });
+    }
+
+    const newContact = await contactsManager.addContact(value);
+
+    res.json({
+      status: "success",
+      code: 201,
+      data: {
+        contact: newContact,
+      },
     });
+  } catch (error) {
+    console.error(error);
+    next(error);
   }
-
-  const newContact = await contactsManager.addContact(name, email, phone);
-
-  res.status(201).json(newContact);
 });
 
 router.delete("/:contactId", async (req, res, next) => {
-  const { contactId } = req.params;
+  try {
+    const { contactId } = req.params;
+
+    const contacts = await contactsManager.listContacts();
 
-  const contacts = await contactsManager.listContacts();
+    const newContacts = await contactsManager.removeContact(contactId);
 
-  const newContacts = await contactsManager.removeContact(contactId);
+    if (newContacts.length === contacts.length) {
+      res.status(404).json({
+        status: "not-found",
+        code: 404,
+        message: "Not found",
+      });
+    }
 
-  if (newContacts.length === contacts.length) {
-    return res.status(404).json({
-      message: "Not found",
+    res.json({
+      status: "success",
+      code: 200,
+      message: `Contact with ID=${contactId} deleted successfully!`,
     });
+  } catch (error) {
+    console.error(error);
+    next(error);
   }
-
-  res.status(200).json({
-    message: `Contact with ID=${contactId} deleted successfully!`,
-  });
 });
 
 router.put("/:contactId", async (req, res, next) => {
-  const { error, value } = schema.validate(req.body);
-  const { contactId } = req.params;
-  const { name, email, phone } = value;
-
-  if (error) {
-    return res.status(400).json({
-      message: error.message,
+  try {
+    const { error, value } = schema.validate(req.body);
+    const { contactId } = req.params;
+    const { name, email, phone } = value;
+
+    if (error) {
+      res.status(400).json({
+        status: "error",
+        code: 400,
+        message: error.message,
+      });
+    }
+
+    if (!name && !email && !phone) {
```

Comment:
To raczej validacja w joi sprawdza czy niema tych pól. I jak już to  OR = "||" a nie AND = "&&"